test/chan.js

```diff
@@ -0,0 +1,69 @@
+var nano = require('..');
+var test = require('tape');
+
+test('adding and removing subscription channels', function (t) {
+
+  //t.plan(7);
+
+  var pub = nano.socket('pub');
+
+  //register some channels: hello, foo, bar
+  var sub = nano.socket('sub', {
+    chan: ['hello', 'foo', 'bar']
+  });
+
+  pub.bind('inproc://filter');
+  sub.connect('inproc://filter');
+
+  //the main object we'll use to establish channel registration and removal
+  var msgs = ['hello world','hello world','hello world','bar world','foo world'];
+  var msgqty = -1; //starting here so first msg counted can be zero
+  var sent = 0;
+
+  sub.on('message', function (buf) {
+
+    var msg = String(buf);
+
+    //increment msgqty when a msg is received (happening about every 100ms)
+    //we'll call finish() on the 5th msg received (case 4, see line 68)
```

Comment:
I wouldn't recommend putting explicit line numbers, it's brittle and prone to breakage as the code changes.